### PR TITLE
Pagination utility to return empty array of current items if data is passed in as array

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?
This PR updates the pagination utility function to review an empty array `[]` when the data passed in is an array but there are no current items. It also updates the function to `useMemo` instead of `useCallback` to avoid an infinite loop that could occur in the `useEffect` `useState` combination.

This fixes an issue with DataTable using `paginate` + `onClickRow` because DataTable relies on data being defaulted to `[]` however usePagination was returning back `undefined` even when the data passed in was an empty array. This ensures that usePagination returns a value that is the same type as the data it received.

#### Where should the reviewer start?
src/js/utils/pagination.js

#### What testing has been done on this PR?
Tested in storybook by adjusting Paginated DataTable with:
```
      <DataTable
        columns={[
          {
            property: "a",
            header: "a"
          },
          {
            property: "b",
            header: "b"
          }
        ]}
        data={[]}
        paginate
        onClickRow={() => {}}
      />
```
Also added jest test for the combination of `paginate` and empty data array to ensure that the pagination controls don't render.

#### How should this be manually tested?
Same as above.

#### Any background context you want to provide?
DataTable render relies on fact that data will default to `[]` but `usePagination` was return data as `undefined` which caused failed render.

#### What are the relevant issues?
Closes https://github.com/grommet/grommet/issues/5123

#### Do the grommet docs need to be updated?
No.

#### Should this PR be mentioned in the release notes?
Fixed error that occurred when `paginate` was used with `onClickRow` on DataTable.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.

### CODE REVIEW

File: src/js/components/DataTable/DataTable.js [hunk 1]
```diff
@@ -406,7 +406,11 @@ const DataTable = ({
           )}
         </StyledDataTable>
       </OverflowContainer>
-      {paginate && items && <Pagination alignSelf="end" {...paginationProps} />}
+      {paginate && items && items.length ? (
+        <Pagination alignSelf="end" {...paginationProps} />
+      ) : (
+        undefined
+      )}
```
Comment (halocline) [inline src/js/components/DataTable/DataTable.js:413]: This would also be valid and result the same:

```suggestion
      {paginate && items && !!items.length && (
        <Pagination alignSelf="end" {...paginationProps} /> 
      )}
```

Or by convention, I think Grommet has tended to use `null` instead of `undefined` in these circumstances.

```suggestion
      {paginate && items && items.length ? (
        <Pagination alignSelf="end" {...paginationProps} />
      ) : null}
```

**Review by halocline — APPROVED** [review]

File: src/js/components/DataTable/DataTable.js (hunk 1 above)
Comment (taysea) [reply src/js/components/DataTable/DataTable.js:413]: Thanks! I'll update to `null` to align with common Grommet convention.

File: src/js/components/DataTable/DataTable.js [hunk 2]
```diff
@@ -406,7 +406,9 @@ const DataTable = ({
           )}
         </StyledDataTable>
       </OverflowContainer>
-      {paginate && items && <Pagination alignSelf="end" {...paginationProps} />}
+      {paginate && items && items.length ? (
```
Comment (ericsoderberghp) [inline src/js/components/DataTable/DataTable.js:409]: I think this might cause a jump in vertical layout if the table has a fixed height but is lazily loaded. I wonder if instead Pagination should treat no items the same a single item, with just one page and disabled controls.

File: src/js/utils/pagination.js [hunk 3]
```diff
@@ -18,24 +18,13 @@ export const usePagination = ({ data, page, step, ...rest }) => {
...
-    [itemsBeginIndex, itemsEndIndex],
-  );
-  const [currentItems, setCurrentItems] = useState(getCurrentItems(data));
 
-  useEffect(() => {
-    if (data) {
-      setCurrentItems(getCurrentItems(data));
+  const currentItems = useMemo(() => {
+    if (Array.isArray(data)) {
+      return data.length ? data.slice(itemsBeginIndex, itemsEndIndex) : [];
```
Comment (ericsoderberghp) [inline src/js/utils/pagination.js:24]: if data is an array of length 0, wont' `data.slice(...)` return `[]` already?

File: src/js/utils/pagination.js [hunk 4]
```diff
@@ -18,24 +18,13 @@ export const usePagination = ({ data, page, step, ...rest }) => {
...
     }
-  }, [data, getCurrentItems, setCurrentItems]);
+    return data;
```
Comment (ericsoderberghp) [inline src/js/utils/pagination.js:26]: Should this instead return `[]` when data isn't an array?

File: src/js/utils/pagination.js (hunk 3 above)
Comment (taysea) [reply src/js/utils/pagination.js:24]: You're right. I think we can remove the check for length now

File: src/js/utils/pagination.js (hunk 4 above)
Comment (taysea) [reply src/js/utils/pagination.js:26]: Yes I think that would be a safe return. No errors were being caused because the only case this return was being called was when List provided `data` as `undefined`, but I think for consistency it would make sense for pagination to return an empty array for currentItems.

File: src/js/components/DataTable/DataTable.js (hunk 2 above)
Comment (taysea) [reply src/js/components/DataTable/DataTable.js:409]: Summarizing offline discussion: Pagination controls should only show if the number of items is great than the step (meaning that there would be more than one page).

**Review by ericsoderberghp — APPROVED** [review]
